compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -242,6 +243,32 @@ fn slice_push_back(
     Ok(Value::Slice(values, typ))
 }
 
+fn str_as_bytes(
+    interner: &NodeInterner,
+    arguments: Vec<(Value, Location)>,
+    location: Location,
+) -> IResult<Value> {
+    let (string, string_location) = check_one_argument(arguments, location)?;
+
+    match string {
+        Value::String(string) => {
+            let string_as_bytes = string.as_bytes();
+            let bytes_vector: Vec<Value> = string_as_bytes.iter().cloned().map(Value::U8).collect();
```

Comment:
Why collect into a `Vec` only to later `.into()` into a `im::Vector`?

Reply:
It didn't like it when I tried to collect directly into an `im::Vector`